pennylane/operation.py

```diff
@@ -534,28 +535,33 @@ def parameters(self):
         """Current parameter values."""
         return self.data.copy()
 
-    @staticmethod
-    def decomposition(*params, wires):
+    @property
+    def hyperparameters(self):
+        """Non-trainable variables that define this operation."""
+        # TODO [Maria/Josh]: this is a placeholder; in future we want to
+        # define hyperparameters for all ops, even if they overwrite init
+        if hasattr(self, "_hyperparameters"):
+            return self._hyperparameters
+        return {}
+
+    def decomposition(self):
         """Defines a decomposition of this operator into products of other operators.
 
-        Args:
-            params (tuple[float, int, array]): operator parameters
-            wires (Union(Sequence[int], Wires)): wires the operator acts on
-
         Returns:
-            list[Operation]
+            list[Operator]
         """
-        raise NotImplementedError
+        return self.compute_decomposition(*self.parameters, self.wires, **self.hyperparameters)
 
-    def decompose(self):
-        """Decomposes this operator into products of other operators.
+    @staticmethod
+    def compute_decomposition(*params, wires, **hyperparams):
+        """
```

Comment:
A docstring and `Returns` is missing here!

Might be good to also have a minimal example?

Reply:
Is it an issue that `*params` precedes `wires`? 🤔 Maybe it isn't, but just curious if there are any odd edge cases